cloud.md

```diff
@@ -0,0 +1,68 @@
+# Cloud features and providers
+
+Part of a complete software ecosystem is the ability for apps to stay in sync. Through preference and/or document transfer users can be editing/reading/playing on one device, move to another and continue where they left off. With a truly cross-platform toolkit like Fyne this should be seamless across all devices.
+
+In addition to the [vision](https://fyne.io/vision/) that *it should be free and simple to develop an application* we must strive to make **usage** of the applications developed under this technology equally democratised. As such any sync solution should, be available to everybody for free or as cheap as possible for the level of functionality delivered.
+
+It is possible that this solution may involve various levels of functionality, for example someone just looking to sync their preferences could perhaps get this for free. However an individual or business working with classified documents that must remain encrypted may wish to pay more for a more sophisticated service.
+
+## Background
+
+The popular platforms now have cloud integration, this is essential in many cases and nice to have in a majority.
+To remain competitive this is an integration we should offer.
+In most cases the cloud is tied to the platform, but as a platform-agnostic toolkit we need to be able to offer a way for the cloud provision to work on any device.
+As a platform that is democratising access to techology we also need to provide choice.
+
+## Architecture / API
+
+## `fyne.io/fyne`
+
+* CloudProvider interface
+    - ProviderName
+    - ProviderIcon
+    - ProviderDescription
+
+* CloudSync interface
+    - EnableCloudPreferences(fyne.App) fyne.Preferences
+
+* CloudStorage interface
+    - EnableCloudStorage(fyne.App) fyne.Storage
+
+* App.SetCloudProvider(CloudProvider) // configure cloud for this app
+
+* App.CloudProvider() CloudProvider // get the (if any) configured provider
+
+## `fyne.io/cloud`
+
+probably a place for lots of docs and information about implementing etc. :)
+
+* CloudProviders() []fyne.CloudProvider
```

Comment:
Hmm this is confusing, how is the dependency order here? `fyne.io/cloud` depends on `fyne.io/fyne` or viceversa? What would be the intended module to be used by the users?

Reply:
The main `fyne.io/fyne` will not depend on this - it provides the shared definitions so that repositories like this can plug in. If developers want to add cloud then they use this repo in addition to Fyne.
What about this line led to the confusion?

Reply:
Oh I see, that make senses. However, I don't understand the purpose of this function. Fyne App can only manage 1 cloud provider right? Then why users could want to register more than one providers in this package (`fyne.io/cloud`)? or this package will allow the users to have something like a wizard to select the desired cloud provider?

Reply:
Oh I see, in that case the function `auto.PresentCloudChoice(fyne.App, fyne.Window)` should be renamed as `auto.ShowCloudSelector` or `auto.ShowCloudSettings`.
IMHO, the name `PresentCloudChoice` is not very clear.

Reply:
Also since we are in the package `fyne.io/cloud`, the function `CloudProviders` could be simplified to `Providers`.

Reply:
I tidied the provider thing up - the `auto` package was over-engineering I think.
If we need to split out because there are too many providers we can solve that later.